### PR TITLE
Modify rule S2252: LaYC format

### PR DESCRIPTION
**Pull request by sebastien-marichal**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM (reviewed for Java)

**Review by alban-auzeill — APPROVED** [review]
Approved the change about Java

File: rules/S2252/why.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+== Why is this an issue?
+
+A {for-link}[`{for-keyword}`] loop is a fundamental programming construct used to execute a block of code repeatedly. However, if the loop's condition is false before the first iteration, the loop will never execute.
```
Comment (Tim-Pohlmann) [inline rules/S2252/why.adoc:3]: Grammarly has some ideas to rewrite this. I do not think it is strictly better. Please judge for yourself which version you like best.
```suggestion
A {for-link}[`{for-keyword}`] loop is a fundamental programming construct used to execute a block of code repeatedly. However, the loop will never run if the loop's condition is false before the first iteration.
```

File: rules/S2252/csharp/rule.adoc [hunk 2]
```diff
@@ -1,39 +1,13 @@
...
-----
-for (int i = 10; i < 10; i++)  // Noncompliant 
-{
-    // ...
-}
-----
+=== Documentation
 
-=== Compliant solution
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/iteration-statements#the-for-statement[The `for` statement]
```
Comment (Tim-Pohlmann) [inline rules/S2252/csharp/rule.adoc:11]: Keep it dry.
```suggestion
* Microsoft Learn - {for-link}[The `for` statement]
```

File: rules/S2252/vbnet/compliant-code.adoc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[source,vbneté,diff-id=1,diff-type=compliant]
```
Comment (Tim-Pohlmann) [inline rules/S2252/vbnet/compliant-code.adoc:1]: `vbneté` :D

File: rules/S2252/vbnet/compliant-code.adoc [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+----
+For i As Integer = 0 To 0 Step 1 ' Compliant: the condition is true at least once, the loop will execute
```
Comment (Tim-Pohlmann) [inline rules/S2252/vbnet/compliant-code.adoc:3]: This is the same loop as in the Noncompliant example.

File: rules/S2252/vbnet/rule.adoc [hunk 5]
```diff
@@ -1,37 +1,13 @@
...
-[source,vbnet]
-----
-For index As Integer = 6 To 5  // Noncompliant 
-    // ...
-Next
-----
+=== Documentation
 
-=== Compliant solution
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/for-next-statement[For...Next Statement (Visual Basic)]
```
Comment (Tim-Pohlmann) [inline rules/S2252/vbnet/rule.adoc:11]:
```suggestion
* Microsoft Learn - {for-link}[For...Next Statement (Visual Basic)]
```

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
For this rule, it makes sense to split it into "Why is this an issue?" and "How to fix it".

File: rules/S2252/why.adoc (hunk 1 above)
Comment (sebastien-marichal) [reply rules/S2252/why.adoc:3]: I would like to keep this order, there is the cause and then the consequence.

File: rules/S2252/why.adoc [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+include::{language}/noncompliant-code.adoc[]
+
+It is important to ensure the loop is executed by fixing the condition to evaluate to `true` at least once:
+
+include::{language}/compliant-code.adoc[]
```
[the remark this answers is not in the document]
Comment (Tim-Pohlmann) [reply rules/S2252/why.adoc:9]: My concern here is that John Cliffton said that "How to fix it" might become the primary tab. If that happens, I would rather have it short and concise and move the "Why?" part to the next tab.

File: rules/S2252/why.adoc [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+This bug has the potential to cause unexpected outcomes such as data loss and data corruption.
```
[the remark this answers is not in the document]
Comment (sebastien-marichal) [reply rules/S2252/why.adoc:11]: I think we should add at least a small sentence on the possible outcomes of this issue. I will try to come up with something less vague.

File: rules/S2252/why.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (sebastien-marichal) [reply rules/S2252/why.adoc]: As discussed, I will keep the single `why` section for this rule.

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**